File: data/test/示例程序.cpp
#include <iostream>
#include <string>

class CleaningRobot {
private:
    std::string model;
    int batteryCapacity;
    bool isCleaning;

public:
    CleaningRobot(std::string model, int batteryCapacity)
        : model(model), batteryCapacity(batteryCapacity), isCleaning(false) {}

    void startCleaning() {
        isCleaning = true;
        std::cout << model << " 开始清扫" << std::endl;
    }

    void stopCleaning() {
        isCleaning = false;
        std::cout << model << " 停止清扫" << std::endl;
    }

    void returnToCharge() {
        std::cout << model << " 返回充电" << std::endl;
    }
};

int main() {
    CleaningRobot robot("SC-2000", 5200);
    robot.startCleaning();
    robot.stopCleaning();
    robot.returnToCharge();
    return 0;
}

File: data/test/示例程序.go
package main

import "fmt"

type CleaningRobot struct {
	model          string
	batteryCapacity int
	isCleaning     bool
}

func NewCleaningRobot(model string, batteryCapacity int) *CleaningRobot {
	return &CleaningRobot{
		model:          model,
		batteryCapacity: batteryCapacity,
		isCleaning:     false,
	}
}

func (r *CleaningRobot) StartCleaning() {
	r.isCleaning = true
	fmt.Printf("%s 开始清扫\n", r.model)
}

func (r *CleaningRobot) StopCleaning() {
	r.isCleaning = false
	fmt.Printf("%s 停止清扫\n", r.model)
}

func (r *CleaningRobot) ReturnToCharge() {
	fmt.Printf("%s 返回充电\n", r.model)
}

func main() {
	robot := NewCleaningRobot("SC-2000", 5200)
	robot.StartCleaning()
	robot.StopCleaning()
	robot.ReturnToCharge()
}
